### PR TITLE
Add changesets configuration to handle versioning and publishing

### PR DESCRIPTION
**Pull request by bidoubiwa**
See #959

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
           node-version: 16
           registry-url: https://registry.npmjs.org/
-      - name: Check release validity
-        run: sh .github/scripts/check-release.sh
-      - name: Check tag format
-        run: sh .github/scripts/check-tag-format.sh "${{ github.event.release.prerelease }}"
```
Comment (brunoocasali) [inline .github/workflows/publish.yml:18]: No need to check the tag format anymore?

File: package.json [hunk 2]
```diff
@@ -31,6 +33,8 @@
     "turbo": "^1.7.0"
   },
   "dependencies": {
+    "@changesets/cli": "^2.26.0",
     "instantsearch.css": "^8.0.0"
-  }
+  },
+  "version": "0.0.0"
```
Comment (brunoocasali) [inline package.json:39]: Can you move to the top? 😄 Also, 0.0.1 O think is more appropriate (even if is going to be eternal) :)

File: packages/instant-meilisearch/scripts/update_version.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const pkg = require('../package.json')
+const fs = require('fs')
+const path = require('path')
+
+const version = pkg.version
+
```
Comment (brunoocasali) [inline packages/instant-meilisearch/scripts/update_version.js:6]:
```suggestion
```

File: packages/instant-meilisearch/scripts/update_version.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+const pckVersion = `export const PACKAGE_VERSION = '${version}'
+`
+
+const p = path.resolve(__dirname, '../src/package-version.ts')
+
+console.log(p)
+fs.writeFileSync(p, pckVersion)
```
Comment (brunoocasali) [inline packages/instant-meilisearch/scripts/update_version.js:13]:
```suggestion
const fileContents = `export const PACKAGE_VERSION = '${version}'
`

const filePath = path.resolve(__dirname, '../src/package-version.ts')

console.log(filePath)
fs.writeFileSync(filePath, fileContents)
```

**Review by brunoocasali — APPROVED** [review]
Just some recommendations :)

File: package.json (hunk 2 above)
Comment (bidoubiwa) [reply package.json:39]: I added `0.0.0` as this is what seems to be the convention in the doc and the repo examples I came across.
- https://turbo.build/repo/docs/reference/codemods#add-package-manager

File: .github/workflows/publish.yml (hunk 1 above)
Comment (bidoubiwa) [reply .github/workflows/publish.yml:18]: I'll maybe add it in the PR that tries the different beta types if it is relevant

**Pull request merged**